README.md

```diff
@@ -373,6 +361,63 @@ interface EnhancedInteger (number) => {
 ```
 
 
+## Composing types
+
+Whenever you want to compose an interface out of several others, use the spread operator for that:
+
+```js
+interface Person {
+  name: Name,
+  birthDate: Number,
+}
+
+interface User {
+  username: String,
+  description?: String,
+  kudos = 0: Number,
+}
+
+interface HumanUser {
+  ...Person,
+  ...User,
+  avatarUrl: String,
+}
+```
+
+You can also use the spread inside object type literals:
+
+```js
+interface Company {
+  name: Name,
+  owner: { ...Person, shareStake: Number },
+}
+```
+
+In case of a name conflict, both properties must be satisfied. Keep that in mind that many types aren’t compatible with one another – for example, `Number` and `String`:
+
+```js
+interface Creature {
+  name: String,
+  character: String,
+  strength: (number) => (number >= 0 && number <= 100),
+}
+
+// This is OK:
+interface Human {
+  ...Creature,
+  name: /^[A-Z][a-z]+$/,
+  character: 'friendly' | 'grumpy',
+}
+
+// This will never be satisfied:
```

Comment:
Yeah, for the user it would be perfect if we detected type collisions statically. But such a promise might be risky on our side as there’s no obvious way to detect some conflicts. Like `a: /^[A-Z]+$/` and `a: /^[a-z]+$/` or `n: (number) => number < 0` vs `n: (number) => number > 10`.